### PR TITLE
feat: add browser <-> nodejs test

### PR DESCRIPTION
**Pull request by diasdavid**
>wip 

- [x] added test
- [x] fix libp2p-websockets https://github.com/libp2p/js-libp2p-websockets/pull/36
- [x] fix secio

### CODE REVIEW

File: test/browser.js [hunk 1]
```diff
@@ -32,12 +32,8 @@ describe('secio browser <-> nodejs', () => {
...
   })
 
-  it('encrypt', (done) => {
-    secio.encrypt(pid, pid._privKey, conn, (err, c) => {
-      expect(err).to.not.exist
-      encryptedConn = c
-      done()
-    })
+  it('encrypt', () => {
+    secio.encrypt(pid, pid._privKey, conn)
```
Comment (diasdavid) [inline test/browser.js]: wait, it doesn't yield a wrapped encrypted conn?

File: README.md [hunk 2]
```diff
@@ -43,11 +43,12 @@ const secio = require('libp2p-secio')
...
 
-### `encrypt(id, key, insecure)`
+### `encrypt(id, key, insecure[, callback])`
 
 - `id: PeerId` - The id of the node.
 - `key: RSAPrivateKey` - The private key of the node.
 - `insecure: PullStream` - The insecure connection.
+- `callback: Function` - Called if an error happens during the initialization.
 
 Returns the `insecure` connection provided, wrapped with secio. This is a pull-stream.
```
Comment (diasdavid) [inline README.md]: This is kind of misleading, saying it returns a 'insecure connection'

**Comment by diasdavid** [comment]
I was just very confused by seeing CI green but locally seeing tests failing. It seems aegir doesn't exist(1) when tests fail in the browser

https://travis-ci.org/libp2p/js-libp2p-secio/jobs/174449763#L1399

**Comment by dignifiedquire** [comment]
> doesn't exist(1) when tests fail in the browser

Yes that is what I had to do, to make Sauce Labs not make all CI red

**Comment by dignifiedquire** [comment]
The underlying reason for the failure to communicate between node.js and the browser is found in https://github.com/libp2p/js-libp2p-crypto/pull/29

**Comment by dignifiedquire** [comment]
Ready for review and passing with https://github.com/libp2p/js-libp2p-crypto/pull/29

**Pull request merged**